base/modulus_remainder.cc

```diff
@@ -0,0 +1,173 @@
+#include "base/modulus_remainder.h"
+
+#include "base/arithmetic.h"
+
+// Based on https://github.com/halide/Halide/blob/main/src/modulus_remainder.cpp.
```

Comment:
Overzealous find-replace on the filename here. Also, when making links like this, I'd get a permalink (one with the hash in it)